RapierPhysics: Add heightfield support.

andytriboletti opened the pull request:
Add rapier terrain physics example

**Description**

I noticed there was an ammo physics terrain example but not rapier.

Does this look ok to contribute?

I love three.js! Thank you!

andytriboletti committed:
Add rapier terrain physics example

andytriboletti committed:
Add screenshot (same as ammo version) and entry to files.json for physics_rapier_terrain

WestLangley commented:
Live demo

https://raw.githack.com/andytriboletti/three.js/dev/examples/physics_rapier_terrain.html

andytriboletti committed:
fix linting error by removing unused variable declaration

Mugen87 commented on examples/physics_rapier_terrain.html:
```diff
@@ -0,0 +1,274 @@
+			import * as THREE from 'three';
+			import Stats from 'three/addons/libs/stats.module.js';
+			import { OrbitControls } from 'three/addons/controls/OrbitControls.js';
+			import RAPIER from '@dimforge/rapier3d';
```
Is it possible to use the existing `RapierPhysics` module to implement this feature? It would be okay to add missing features to it if necessary. Something like:
```js
physics = await RapierPhysics();
physics.addHeightfield( ... );
````
```

andytriboletti replied:
Would you want me to update the Rapier version to the latest as well? Would that be a new PR? It's a few versions behind in RapierPhysics.

andytriboletti replied:
I was able to use existing RapierPhysics.

Mugen87 replied:
> Would that be a new PR?

Yes, let's focus this PR on the terrain topic.

andytriboletti committed:
Responding to PR feedback

Use RapierPhysics instead of direct Rapier in example. Update RapierPhysics with addHeightfield.

andytriboletti committed:
Fix CodeQL notice - remove unused function

Mugen87 committed:
Update physics_rapier_terrain.html

Clean up code style.

Mugen87 committed:
Update RapierPhysics.js

Mugen87 commented on examples/jsm/physics/RapierPhysics.js:
```diff
@@ -31,14 +31,14 @@ function getShape( geometry ) {
 
 		const radius = parameters.radiusBottom !== undefined ? parameters.radiusBottom : 0.5;
-		const length = parameters.length !== undefined ? parameters.length : 0.5;
+		const length = parameters.height !== undefined ? parameters.height : 0.5;
```
While testing I have notice the collisions for the cylinder meshes were incorrect. Turns out the wrong parameters were evaluated.

Mugen87 committed:
Update RapierPhysics.js
